Mathlib/Analysis/Matrix.lean

```diff
@@ -185,6 +190,35 @@ theorem norm_diagonal [DecidableEq n] (v : n → α) : ‖diagonal v‖ = ‖v
 instance [Nonempty n] [DecidableEq n] [One α] [NormOneClass α] : NormOneClass (Matrix n n α) :=
   ⟨(norm_diagonal _).trans <| norm_one⟩
 
+namespace Int
+
+open Finset
+
+/-- The norm of an integral matrix is the cast of a natural number -/
+lemma norm_eq_NatCast (A : Matrix m n ℤ) : ∃ (a : ℕ), ‖A‖ = ↑a := by
+  use sup univ fun i ↦ sup univ fun j ↦ (A i j).natAbs
+  rw [norm_eq_sup_sup_nnnorm, ← NNReal.coe_natCast, NNReal.coe_inj,
+    comp_sup_eq_sup_comp_of_is_total Nat.cast Nat.mono_cast
+    (by simp only [bot_eq_zero', CharP.cast_eq_zero])]
+  congr
+  rw [Function.comp_def]
+  ext; congr
+  rw [comp_sup_eq_sup_comp_of_is_total Nat.cast Nat.mono_cast
+     (by simp only [bot_eq_zero', CharP.cast_eq_zero])]
+  congr; ext; congr;
+  rw [Function.comp_apply, NNReal.natCast_natAbs]
+
+/-- The norm of a non-singular integral matrix is a positive natural number-/
+lemma one_le_norm_of_nonzero (A : Matrix m n ℤ) (hA_nezero : A ≠ 0) (a : ℕ)
```

Comment:
Isn't it more natural to add a `def natNorm : ℕ` that is defined as what you currently have on line 199 and then restate `norm_eq_NatCast` as `norm_eq_natNorm`, so that you can then state this lemma as `one_le_natNorm` showing that the `natNorm` of a non-zero matrix is at least one?

Reply:
I implemented your suggestions. It makes more sense.

Thanks for the advice!